File: src/main/java/org/funivan/lologa/algo/ga/genome/value/RemovedTilesValue.java
package org.funivan.lologa.algo.ga.genome.value;

import org.funivan.lologa.tiles.TilesInterface;

public class RemovedTilesValue implements ValueInterface {
    @Override
    public String type() {
        return "removed_tiles";
    }

    @Override
    public Double value(TilesInterface original, TilesInterface tiles) {
        return 1 - (double) tiles.size() / (double) original.size();
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/value/RemovedTilesValue.kt
package org.funivan.lologa.algo.ga.genome.value

import org.funivan.lologa.tiles.TilesInterface

class RemovedTilesValue : ScoreInterface {
    override fun type(): String {
        return "removed_tiles"
    }

    override fun value(original: TilesInterface, tiles: TilesInterface): Double {
        return 1 - tiles.size().toDouble() / original.size().toDouble()
    }
}
